sandbox/src/main/java/org/robolectric/internal/bytecode/ClassInstrumentor.java

```diff
@@ -179,6 +183,56 @@ public void instrument(MutableClass mutableClass) {
     }
   }
 
+  // See https://github.com/robolectric/robolectric/issues/6840
+  // Adds Set(int, object) to android.util.SparseArray.
+  private void addSetToSparseArray(MutableClass mutableClass) {
+    for (MethodNode method : mutableClass.getMethods()) {
+      if ("set".equals(method.name)) {
```

Comment:
Does it need to check method parameters too?

Reply:
Older versions don't have a set method at all, so it's really just checking if one exists in the first place as then the bug with the kotlin [] operators won't happen. It's probably a good idea to check the whole signature though just in case so we can push it in the next sync.